app/controllers/users_controller.rb

```diff
@@ -1309,6 +1310,38 @@ def read_faq
     render json: success_json
   end
 
+  def recent_searches
+    if !SiteSetting.log_search_queries
+      return render json: failed_json.merge(
+                      error: I18n.t("user_activity.no_log_search_queries")
+                    )
```

Comment:
I know we have `failed_json` since a long time ago but I've always found returning a `200` response with an error message a very odd pattern to follow. It'll be great if we can change this pattern if it isn't too hard.